engine/runner/src/main/scala/org/enso/runner/LanguageServerApp.scala

```diff
@@ -18,12 +19,24 @@ object LanguageServerApp {
     * @param config a config
     * @param logLevel log level
     */
-  def run(config: LanguageServerConfig, logLevel: LogLevel): Unit = {
+  def run(
+    config: LanguageServerConfig,
+    logLevel: LogLevel,
+    deamonize: Boolean
+  ): Unit = {
     val server = new LanguageServerComponent(config, logLevel)
     Runtime.getRuntime.addShutdownHook(new Thread(() => {
       Await.result(server.stop(), 20.seconds)
     }))
     Await.result(server.start(), 20.seconds)
+    if (deamonize) {
+      val lock = new AnyRef
+      lock.synchronized {
+        lock.wait()
```

Comment:
Shouldn't this be a while loop?

A `wait` can theoretically wake-up spuriously.

Reply:
It will wake up only if you call `lock.notify/notifyAll` or `interrupt` on the main thread. Therefore, it will never happen.